src/plugins/dev_tools/public/application.tsx

```diff
@@ -57,7 +63,25 @@ function DevToolsWrapper({ devTools, activeDevTool, updateRoute }: DevToolsWrapp
             }}
           >
             <EuiToolTip content={currentDevTool.tooltipContent}>
-              <span>{currentDevTool.title}</span>
+              <span>
+                <EuiFlexGroup gutterSize="s" alignItems="center" responsive={false}>
+                  <EuiFlexItem grow={false}>{currentDevTool.title}</EuiFlexItem>
+
+                  {currentDevTool.isBeta && (
+                    <EuiFlexItem grow={false}>
+                      <EuiBetaBadge
+                        label={i18n.translate('devTools.badge.betaLabel', {
+                          defaultMessage: 'Beta',
+                        })}
+                        tooltipContent={i18n.translate('devTools.badge.betaTooltipText', {
+                          defaultMessage:
+                            'This feature might change drastically in future releases',
+                        })}
+                      />
+                    </EuiFlexItem>
+                  )}
+                </EuiFlexGroup>
+              </span>
```

Comment:
If the purpose for this `EuiFlexGroup` structure is just to vertically center the `EuiBetaBadge` component with the preceding text, you may be able to simplify your markup this by getting rid of these wrapping element entirely and just applying a custom `vertical-align` style to the `EuiBetaBadge` to override the default `super` style. For example:

```suggestion
              {currentDevTool.title}{' '}
              {currentDevTool.isBeta && (
                <EuiBetaBadge
                  className="devApp__tabBeta"
                  label={i18n.translate('devTools.badge.betaLabel', {
                    defaultMessage: 'Beta',
                  })}
                  size="s"
                  tooltipContent={i18n.translate('devTools.badge.betaTooltipText', {
                    defaultMessage:
                      'This feature might change drastically in future releases',
                  })}
                />
              )}
```

And then just apply your `vertical-align` styles to that `.devApp__tabBeta` class.

Reply:
Makes a ton of sense! I've just made the changes and updated the screenshot in the PR description